control/plugin/cpolicy/tree.go

```diff
@@ -140,28 +136,23 @@ func (c *ConfigPolicy) Get(ns []string) *ConfigPolicyNode {
 		return &t
 	default:
 		return t.(*ConfigPolicyNode)
-
 	}
 }
 
-func (c *ConfigPolicy) GetAll() map[string]*ConfigPolicyNode {
-	// Automatically freeze on first Get
-	if !c.config.Frozen() {
-		c.config.Freeze()
-	}
+type keyNode struct {
+	Key  []string
+	Node *ConfigPolicyNode
```

Comment:
Yes, but it also allows:

`for _, rule := range node.RulesAsTable() {` 
instead of 
`for _, rule := range node.Node.RulesAsTable() {`

and 

`for _, cpolicy := range lPlugin.ConfigPolicy.GetAll() {` 
instead of 
```
for _, kn := range lPlugin.ConfigPolicy.GetAll() {
	cpolicy := kn.Node
```